awx/sso/backends.py

```diff
@@ -383,31 +381,73 @@ def on_populate_user(sender, **kwargs):
+            logger.debug("Team named {} in LDAP team map settings is invalid due to missing organization".format(team_name))
+            continue
+        # A team may have an org that we didn't previously evaluate
+        if team_opts['organization'] not in existing_org_names:
+            Organization.objects.get_or_create(name=team_opts['organization'])
+            # Append it to the list so that we don't try and create this again if its used by another team
+            existing_org_names.append(team_opts['organization'])
+        if team_name not in existing_team_names:
+            # Going a little inefficient here, we could prob get name/ids above and then set this by ID but I've got limited time r/n
+            #    and this code should only really be getting hit once if an LDAP config change adds a team with an org not built.
+            org, created = Organization.objects.get_or_create(name=team_opts['organization'])
+            Team.objects.get_or_create(name=team_name, organization=org)
+    # End move some day
+    # ==============================================================================================================
+
+    # Compute in memory what the state is of the different LDAP orgs
+    desired_org_states = {}
     for org_name, org_opts in org_map.items():
-        org, created = Organization.objects.get_or_create(name=org_name)
         remove = bool(org_opts.get('remove', True))
         admins_opts = org_opts.get('admins', None)
         remove_admins = bool(org_opts.get('remove_admins', remove))
-        _update_m2m_from_groups(user, ldap_user, org.admin_role.members, admins_opts, remove_admins)
+        desired_org_states[org_name] = {}
+        desired_org_states[org_name]['admin_role'] = _update_m2m_from_groups(ldap_user_group_dns, admins_opts, remove_admins)
         auditors_opts = org_opts.get('auditors', None)
         remove_auditors = bool(org_opts.get('remove_auditors', remove))
-        _update_m2m_from_groups(user, ldap_user, org.auditor_role.members, auditors_opts, remove_auditors)
+        desired_org_states[org_name]['auditor_role'] = _update_m2m_from_groups(ldap_user_group_dns, auditors_opts, remove_auditors)
         users_opts = org_opts.get('users', None)
         remove_users = bool(org_opts.get('remove_users', remove))
-        _update_m2m_from_groups(user, ldap_user, org.member_role.members, users_opts, remove_users)
+        desired_org_states[org_name]['member_role'] = _update_m2m_from_groups(ldap_user_group_dns, users_opts, remove_users)
 
-    # Update team membership based on group memberships.
-    team_map = getattr(backend.settings, 'TEAM_MAP', {})
+        # If everything returned None (because there was no configuration) we can remove this org from our map
+        if (
+            desired_org_states[org_name]['admin_role'] == None
+            and desired_org_states[org_name]['auditor_role'] == None
+            and desired_org_states[org_name]['member_role'] == None
```

Comment:
I'd prefer to tighten this up, like `if all(desired_org_states[org_name][role_field] is None for role_field in roles):`

I also thought that the linter got you in trouble for `== None`.

where the definition of `roles = ['admin_role', 'auditor_role', 'member_role']` is shared with its use later on for saving the data.